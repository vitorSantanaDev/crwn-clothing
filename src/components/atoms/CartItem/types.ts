import { IProductCartItem } from 'interfaces'

export type ICartItemProps = IProductCartItem
