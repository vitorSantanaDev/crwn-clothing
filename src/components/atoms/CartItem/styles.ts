import styled from 'styled-components'

export const CartItemWrapper = styled.div`
  width: 100%;
  display: flex;
  height: 80px;
  margin-bottom: 15px;
`

export const InfosItemWrapper = styled.div`
  width: 70%;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
  padding: 10px 20px;
`

export const ItemName = styled.h2`
  font-size: 16px;
`

export const ItemQuantity = styled.span`
  font-size: 16px;
`

export const ImageItem = styled.img`
  width: 30%;
`
