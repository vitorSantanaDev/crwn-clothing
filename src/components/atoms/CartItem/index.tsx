import { ICartItemProps } from './types'

import * as S from './styles'

export default function CartItem({
  name,
  quantity,
  imageUrl,
  price
}: ICartItemProps) {
  return (
    <S.CartItemWrapper>
      <S.ImageItem src={imageUrl} alt={name} />
      <S.InfosItemWrapper>
        <S.ItemName>{name}</S.ItemName>
        <S.ItemQuantity>
          {quantity} x ${price}
        </S.ItemQuantity>
      </S.InfosItemWrapper>
    </S.CartItemWrapper>
  )
}
