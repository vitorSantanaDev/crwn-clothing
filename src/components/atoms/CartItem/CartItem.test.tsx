import { render, screen } from '@testing-library/react'
import SHOP_DATA from 'database/shop-data'

import CartItem from '.'

const mockData = SHOP_DATA[0].items[0]

describe('<CartItem />', () => {
  it('should render cart item correctly', () => {
    render(<CartItem quantity={3} {...mockData} />)

    const imageItem = screen.getByRole('img', { name: mockData.name })
    const nameItem = screen.getByText(mockData.name)
    const priceAndQuantity = screen.getByText(`${3} x $${mockData.price}`)

    expect(imageItem).toBeInTheDocument()
    expect(nameItem).toBeInTheDocument()
    expect(priceAndQuantity).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = render(<CartItem quantity={3} {...mockData} />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
