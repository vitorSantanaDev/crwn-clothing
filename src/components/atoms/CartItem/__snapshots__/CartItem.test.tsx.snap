// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartItem /> should match snapshot 1`] = `
<div
  class="sc-bczRLJ ggPchI"
>
  <img
    alt="Brown Brim"
    class="sc-eCYdqJ isuBik"
    src="https://i.ibb.co/ZYW3VTp/brown-brim.png"
  />
  <div
    class="sc-gsnTZi cTyJxh"
  >
    <h2
      class="sc-dkzDqf bnRgTK"
    >
      Brown Brim
    </h2>
    <span
      class="sc-hKMtZM jFHkdN"
    >
      3
       x $
      25
    </span>
  </div>
</div>
`;
